### PR TITLE
Use ENS name as account identifier if it exists

### PR DESCRIPTION
**Pull request by jackmpcollins**
Show ENS name instead of truncated address in transactions pill if one exists for the address.

Before
Screen Shot 2021-12-12 at 12 04 05 AM

After
Screen Shot 2021-12-12 at 12 03 50 AM

### CODE REVIEW

File: packages/frontend/src/components/header/TxPill.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React, { useEffect, useState } from 'react'
+import { ethers } from 'ethers';
```
Comment (kangarang) [inline packages/frontend/src/components/header/TxPill.tsx:2]: i believe this line can be removed (extraneous)

File: packages/frontend/src/components/header/TxPill.tsx [hunk 2]
```diff
@@ -44,6 +46,12 @@ const TxPill = () => {
     }
   }, [transactions])
 
+  useEffect(() => {
+    if (address) {
+      provider?.lookupAddress(address.toString()).then(setENSName);
+    }
+  })
```
Comment (kangarang) [inline packages/frontend/src/components/header/TxPill.tsx:53]: add useEffect dependencies `[address, provider]`

**Review by kangarang — CHANGES_REQUESTED** [review]
hi @jackmpcollins, thanks for this pr. currently, these changes will only work if the user is connected to ethereum mainnet. ideally, we should display the user's ens name regardless of connected network. instead of using the connected provider, i recommend grabbing the mainnet provider via the util [`getProviderByNetworkName('ethereum')`](https://github.com/hop-protocol/hop/blob/develop/packages/frontend/src/utils/getProvider.ts#L17) and using that to resolve the ens name.

**Comment by jackmpcollins** [comment]
Thanks for the review @kangarang . Updated now.

**Review by kangarang — APPROVED** [review]
LGTM. thanks again for your contribution @jackmpcollins

**Pull request merged**